wandb/sdk/internal/internal_api.py

```diff
@@ -1040,13 +1037,64 @@ def create_run_queue(
         ]
         return result
 
+    @normalize_exceptions
+    def push_to_run_queue_by_name(
+        self, entity: str, project: str, queue_name: str, run_spec: str
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Queryless mutation, should be used before legacy fallback method
+        """
+
+        mutation = gql(
+            """
+        mutation pushToRunQueueByName(
+            $entityName: String!,
+            $projectName: String!,
+            $queueName: String!,
+            $runSpec: JSONString!,
+        ) {
+            pushToRunQueueByName(
+                input: {
+                    entityName: $entityName,
+                    projectName: $projectName,
+                    queueName: $queueName,
+                    runSpec: $runSpec
+                }
+            ) {
+                runQueueItemId
+            }
+        }
+        """
+        )
+        variables = {
+            "entityName": entity,
+            "projectName": project,
+            "queueName": queue_name,
+            "runSpec": run_spec,
+        }
+
+        result: Optional[Dict[str, Any]] = self.gql(mutation, variables).get(
+            "pushToRunQueueByName"
+        )
+        return result
+
     @normalize_exceptions
     def push_to_run_queue(
         self, queue_name: str, launch_spec: Dict[str, str]
     ) -> Optional[Dict[str, Any]]:
-        # TODO(kdg): add pushToRunQueueByName to avoid this extra query
         entity = launch_spec["entity"]
         project = launch_spec["project"]
+        run_spec = json.dumps(launch_spec)
+
+        push_result = self.push_to_run_queue_by_name(
+            entity, project, queue_name, run_spec
+        )
+
+        if push_result:
+            return push_result
+
+        """ Legacy Method """
+        wandb.termlog("Using legacy run queue push.")
```

Comment:
Lets remove this, users shouldn't care whether they're using legacy push or not

Reply:
Hmm... why did you mark this as resolved? Doesn't seem to be addressed

Reply:
Weird, must have snuck back in. Was definitely removed lol.